code/modules/mob/mob.dm

```diff
@@ -1347,3 +1347,151 @@
+
+		if("animal")
+			var/path = pick(/mob/living/simple_animal/hostile/carp,
+							/mob/living/simple_animal/hostile/bear,
+							/mob/living/simple_animal/hostile/mushroom,
+							/mob/living/simple_animal/hostile/statue,
+							/mob/living/simple_animal/hostile/retaliate/bat,
+							/mob/living/simple_animal/hostile/retaliate/goat,
+							/mob/living/simple_animal/hostile/killertomato,
+							/mob/living/simple_animal/hostile/giant_spider,
+							/mob/living/simple_animal/hostile/giant_spider/hunter,
+							/mob/living/simple_animal/hostile/blob/blobbernaut/independent,
+							/mob/living/simple_animal/hostile/carp/ranged,
+							/mob/living/simple_animal/hostile/carp/ranged/chaos,
+							/mob/living/simple_animal/hostile/asteroid/basilisk/watcher,
+							/mob/living/simple_animal/hostile/asteroid/goliath/beast,
+							/mob/living/simple_animal/hostile/headcrab,
+							/mob/living/simple_animal/hostile/morph,
+							/mob/living/simple_animal/hostile/stickman,
+							/mob/living/simple_animal/hostile/stickman/dog,
+							/mob/living/simple_animal/hostile/megafauna/dragon/lesser,
+							/mob/living/simple_animal/hostile/gorilla,
+							/mob/living/simple_animal/parrot,
+							/mob/living/simple_animal/pet/dog/corgi,
+							/mob/living/simple_animal/crab,
+							/mob/living/simple_animal/pet/dog/pug,
+							/mob/living/simple_animal/pet/cat,
+							/mob/living/simple_animal/mouse,
+							/mob/living/simple_animal/chicken,
+							/mob/living/simple_animal/cow,
+							/mob/living/simple_animal/hostile/lizard,
+							/mob/living/simple_animal/pet/fox,
+							/mob/living/simple_animal/butterfly,
+							/mob/living/simple_animal/pet/cat/cak,
+							/mob/living/simple_animal/chick)
+			new_mob = new path(loc)
+
+		if("humanoid")
+			var/mob/living/carbon/human/new_human = new (loc)
+
+			if(prob(50))
+				var/list/chooseable_races = list()
+				for(var/speciestype in subtypesof(/datum/species))
+					var/datum/species/S = speciestype
+					if(initial(S.changesource_flags) & WABBAJACK)
+						chooseable_races += speciestype
+
+				if(length(chooseable_races))
+					new_human.set_species(pick(chooseable_races))
+
+			// Randomize everything but the species, which was already handled above.
+			new_human.randomize_human_appearance(~RANDOMIZE_SPECIES)
+			new_human.update_body()
+			new_human.update_hair()
+			new_human.update_body_parts()
+			new_human.dna.update_dna_identity()
+			new_mob = new_human
+
+	if(!new_mob)
+		return
+
+	// Some forms can still wear some items
+	for(var/obj/item/W in contents)
+		new_mob.equip_to_appropriate_slot(W)
+
+	log_message("became [new_mob.name]([new_mob.type])", LOG_ATTACK, color="orange")
+	new_mob.set_combat_mode(TRUE)
+	wabbajack_act(new_mob)
+	to_chat(new_mob, span_warning("Your form morphs into that of a [randomize]."))
+
+	var/poly_msg = get_policy(POLICY_POLYMORPH)
+	if(poly_msg)
+		to_chat(new_mob, poly_msg)
+
+	transfer_observers_to(new_mob)
+
+	qdel(src)
+	return new_mob
```

Comment:
```suggestion
	. = new_mob
	qdel(src)
```
Just in case, always leave deleting source as the last action.

Reply:
3d20794272aeecd23501f784839f8315561c3e67